galgebra/metric.py

```diff
@@ -202,22 +200,12 @@ def apply(expr):
                 obj += coef * base
         else:
             for (coef, base) in zip(coefs, bases):
-                obj += modes(coef) * base
+                obj += Simp.modes(coef) * base
         return obj
 
     @staticmethod
     def applymv(mv):
-        (coefs, bases) = linear_expand(mv.obj)
-        obj = S(0)
-        if isinstance(Simp.modes, list) or isinstance(Simp.modes, tuple):
-            for (coef, base) in zip(coefs, bases):
-                for mode in Simp.modes:
-                    coef = mode(coef)
-                obj += coef * base
-        else:
-            for (coef, base) in zip(coefs, bases):
-                obj += modes(coef) * base
-        mv.obj = obj
+        mv.obj = Simp.apply(mv.obj)
```

Comment:
The code was repetitive, so calls the method with exactly the same semantic instead.